src/Components/Authentication/signup.js

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect, useState } from 'react'
+import { Link } from 'react-router-dom'
+import { useAuth } from '../../Context/AuthContext/authContext'
+import './authentication.css'
+
+const Signup = () => {
+  const [userDetails, setUserDetails] = useState({firstName: '',lastName: '',email: '', password: '', confirmPassword: '', terms: false})
+  const [toggleShowPassword, setToggleShowPassword] = useState(true)
+  const [toggleShowConfirmPass, setToggleShowConfirmPass] = useState(true)
+  const [error, setError] = useState({isError: false, text: ''})
```

Comment:
suggestion : you could use the useReducer

Reply:
Okayy I will replace it with useReducer. Thankyou!!